airbyte-integrations/connectors/source-iterable/source_iterable/auth.py

```diff
@@ -0,0 +1,39 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from typing import Any, Mapping
+
+from base_python import HttpAuthenticator
+
+
+class ParamsAuthenticator(HttpAuthenticator):
+    def __init__(self, api_key: str, auth_name: str = "api_key"):
+        self.auth_name = auth_name
+        self._api_key = api_key
+
+    def get_auth_header(self) -> Mapping[str, Any]:
+        return {}
+
+    def get_auth_params(self) -> Mapping[str, Any]:
```

Comment:
if auth is not in the headers, is it worth creating an entire `Authenticator` class? Why don't we just pass the token in the constructor parameters for each stream and then the base `IterableStream` class can add it in the `request_params` method? `authenticator` is an optional parameter for HTTP streams anyways.

Reply:
yeah, good idea, updated